### ISSUE TITLE
ceph-ansible tries to create new OSDs on already provisioned devices and fails

### ISSUE THREAD

**Issue by mitt-fn** [ceph/ceph-ansible#5320]
**Bug Report**

What happened:
Letting ceph-ansible run over an already deployed cluster that is containerized fails, because it is trying to provision new OSDs on already provisioned devices using ceph-volume batch. I was not able to reproduce this in a non-containerized setting.

What you expected to happen:
Already provisioned devices are not tried to be provisioned again.

How to reproduce it (minimal and precise):

Create a containerized cluster using ceph-ansible without autodiscovery. Give at least one device and one dedicated_device. Execute ceph-ansible again with the same settings.

inventory: https://gist.github.com/mitt-fn/963bb68099479f745659a81974ac9310
all.yml: https://gist.github.com/mitt-fn/a74c0ca09b4c6eaa8512ba0b79e27aa7
clients.yml: https://gist.github.com/mitt-fn/bbfe04e383d1e3d11a1b24f6ce930660
mgrs.yml: https://gist.github.com/mitt-fn/ab4554bcbaf335ff4c70a72f73751bb1
osds.yml: https://gist.github.com/mitt-fn/c7e4656ee417ec50309537604e5e68c3 Note: devices are defined using host_vars.
example osds.yml for a host: https://gist.github.com/mitt-fn/e627cd9ba12bd5a605232686b604500f

ansible.log: https://gist.github.com/mitt-fn/6c8b5e60f3ad7c6b978b1e9891da2133

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.4 LTS
* Kernel (e.g. `uname -a`): Linux ceph-mon1 5.3.0-42-generic #34~18.04.1-Ubuntu SMP Fri Feb 28 13:42:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Docker version if applicable (e.g. `docker version`): Docker version 19.03.6, build 369ce74a3c
* Ansible version (e.g. `ansible-playbook --version`):  ansible-playbook 2.9.5
* ceph-ansible version (e.g. `git head or tag or stable branch`): 7acd9686 (stable-4.0 branch pulled in as submodule)
* Ceph version (e.g. `ceph -v`): ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable)

**Comment by mitt-fn** [comment]
After playing around a bit more, I've noticed that I only run into this when there are dedicated_devices declared. I've also tested using the standard /dev/sd* notation in the devices list, but that doesn't seem to have an effect on this bug.

**Comment by dsavineau** [comment]
> Letting ceph-ansible run over an already deployed cluster that is containerized fails, because it is trying to provision new OSDs on already provisioned devices using ceph-volume batch. I was not able to reproduce this in a non-containerized setting.

I'm not sure to correctly understand the workflow.

Did you try first to deploy your containerized environment with success and then rerun the same playbook without any changes ?

**Comment by mitt-fn** [comment]
> Did you try first to deploy your containerized environment with success and then rerun the same playbook without any changes ?

Initially, I wanted to add a ganesha NFS service to an already deployed Ceph cluster, but the deployment did not succeed because of the failure that's happening on the OSD hosts. I then reverted the changes I made and let the playbook run with the same settings the cluster was initially deployed with and ran into the same failures.
To make sure it didn't have anything to do with the NFS configuration, I reproduced the error on my laptop using VMs without trying to deploy any NFS services so I didn't mention it in my initial bug report.
